101-app-configuration/azuredeploy.json

```diff
@@ -0,0 +1,188 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+      "appConfigStoreName": {
+        "type": "string",
+        "metadata": {
+          "description": "App configuration store name."
+        }
+      },
+      "adminPassword": {
+        "type": "securestring",
+        "metadata": {
+          "description": "Password for the Virtual Machine."
+        }
+      },
+      "location": {
+        "type": "string",
+        "defaultValue": "[resourceGroup().location]",
+        "metadata": {
+          "description": "Location for all resources."
+        },
+        "vmSize": {
+          "type": "string",
+          "defaultValue": "Standard_DS1_v2",
+          "metadata": {
+              "description": "The size of the VM"
+          }
+        }
+      }
+    },
+    "variables": {
+      "storageAccountName": "[concat(uniquestring(resourceGroup().id), 'sawinvm')]",
+      "nicName": "myVMNic",
+      "addressPrefix": "10.0.0.0/16",
+      "subnetName": "Subnet",
+      "subnetPrefix": "10.0.0.0/24",
+      "publicIPAddressName": "myPublicIP",
+      "vmName": "SimpleWinVM",
+      "virtualNetworkName": "MyVNET",
+      "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]",
+      "appConfigRef": "[resourceId('Microsoft.AppConfiguration/configurationStores', parameters('appConfigStoreName'))]",
+      "appConfigApiVersion": "2019-02-01-preview",
```

Comment:
remove this and use literal apiVersions where needed

Reply:
why shouldn't this be a variable?

Reply:
it's an anti-pattern - the apiVersion affects the shape of the request so we require literals